libminifi/include/utils/TestUtils.h

```diff
@@ -0,0 +1,80 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string>
+
+#include "../../test/TestBase.h"
+
+#ifdef WIN32
+#include <direct.h>
+#define GetCurrentDir _getcwd
+#else
+#include <unistd.h>
+#define GetCurrentDir getcwd
+#endif
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace utils {
+
+std::string getCurrentWorkingDir(void) {
+  char buff[FILENAME_MAX];
+  GetCurrentDir(buff, FILENAME_MAX);
+  std::string current_working_dir(buff);
+  return current_working_dir;
+}
+
+std::string createTempDir(TestController* testController) {
+  char dirtemplate[] = "/tmp/gt.XXXXXX";
+  std::string temp_dir = testController->createTempDirectory(dirtemplate);
+  REQUIRE(!temp_dir.empty());
+  struct stat buffer;
+  REQUIRE(-1 != stat(temp_dir.c_str(), &buffer));
+  REQUIRE(S_ISDIR(buffer.st_mode));
+  return temp_dir;
+}
+
+std::string putFileToDir(const std::string& dir_path, const std::string& file_name, const std::string& content) {
+  std::string file_path(dir_path + utils::file::FileUtils::get_separator() + file_name);
+  std::ofstream out_file(file_path);
```

Comment:
I would write these files in binary mode.  In text mode (which is the default), the contents of the file will be different in Windows and Linux/Mac because of line endings, which will complicate some tests.

Reply:
Updated.